controller/src/plan.rs

```diff
@@ -20,10 +24,28 @@ impl ControllerPlan {
         let nats = config.nats.connect_with_retry().await?;
 
         let scheduler_plan = config.scheduler.map(|_| SchedulerPlan);
-        let dns_plan = config.dns.map(|options| DnsPlan {
-            options,
-            nc: nats.clone(),
-        });
+        let dns_plan = if let Some(options) = config.dns {
```

Comment:
Minor: would using a `map` here make it slightly easier to read? Only one if/else instead of nested if/elses?

Reply:
Good instinct! But in this case the `if/else` is more readable because I use the `?` operator within it. `?` provides a short-circuit return for the function it is run inside, so if you use `?` within a closure passed to a map, it doesn't short-circuit return for the outer method, just the closure, which you then need to pop up again. It gets to be kind of messy and, though a bit more verbose, I prefer just using `if let`/`else` in those situations.